keras/utils/timeseries_dataset.py

```diff
@@ -189,41 +208,21 @@ def timeseries_dataset_from_array(
   num_seqs = end_index - start_index - (sequence_length * sampling_rate) + 1
   if targets is not None:
     num_seqs = min(num_seqs, len(targets))
-  if num_seqs < 2147483647:
-    index_dtype = 'int32'
-  else:
-    index_dtype = 'int64'
-
-  # Generate start positions
-  start_positions = np.arange(0, num_seqs, sequence_stride, dtype=index_dtype)
-  if shuffle:
-    if seed is None:
-      seed = np.random.randint(1e6)
-    rng = np.random.RandomState(seed)
-    rng.shuffle(start_positions)
-
-  sequence_length = tf.cast(sequence_length, dtype=index_dtype)
-  sampling_rate = tf.cast(sampling_rate, dtype=index_dtype)
-
-  positions_ds = tf.data.Dataset.from_tensors(start_positions).repeat()
-
-  # For each initial window position, generates indices of the window elements
-  indices = tf.data.Dataset.zip(
-      (tf.data.Dataset.range(len(start_positions)), positions_ds)).map(
-          lambda i, positions: tf.range(  # pylint: disable=g-long-lambda
-              positions[i],
-              positions[i] + sequence_length * sampling_rate,
-              sampling_rate),
-          num_parallel_calls=tf.data.AUTOTUNE)
-
-  dataset = sequences_from_indices(data, indices, start_index, end_index)
+
+  dataset = tf.data.Dataset.from_tensor_slices(data[start_index : end_index])
+  dataset = dataset.window(sequence_length, shift=sequence_stride, stride=sampling_rate, drop_remainder=True)
+  dataset = dataset.flat_map(lambda x: x.batch(sequence_length, drop_remainder=True))
```

Comment:
I am bit confused about why we need this here.

Reply:
I think it's good. I do all unit tests. See the tutorial to TF Dataset.

`dataset = dataset.flat_map(lambda x: x.batch(sequence_length, drop_remainder=True))` = this part with `drop_remainder=True` makes the issue #16315 solved because XLA makes shape static and is defined in all axis !